Solutions/Cyware/Hunting queries/DetectingSuspiciousPowerShellCommandExecutions.yaml

```diff
@@ -0,0 +1,34 @@
+id: 47559078-dc4c-4de3-96fe-270d4ca95446
+name: Detecting Suspicious PowerShell Command Executions
+description: |
+  'Query identifies users denied registration for multiple webinars or recordings but successfully registered for at least one event. Threshold variable adjusts number of events user needs to be rejected from.'
+description_detailed: |
+  'Identify suspicious PowerShell command line executions that match indicators from a watchlist. This can help in uncovering scripts or commands related to malware execution or lateral movement.'
+requiredDataConnectors: [
+'DeviceProcessEvents'
+]
+tactics:
+  - Execution
+relevantTechniques:
+  - T1059
+query: |
+DeviceProcessEvents
+| where ProcessCommandLine has "powershell.exe"
+// Add a placeholder column to facilitate the join
+| extend dummyKey = 1
+| join kind=inner (
+    _GetWatchlist('CTIX')
+    | where isnotempty(SearchKey)
+    // Add the same placeholder column here
+    | extend dummyKey = 1
+) on dummyKey
+// Now, filter the joined results using the contains operator
+| where ProcessCommandLine contains SearchKey
+| project TimeGenerated, DeviceName, InitiatingProcessCommandLine, ProcessCommandLine, SearchKey, LastUpdatedTimeUTC
+
+version: 1.0.0
+metadata:
```

Comment:
Hi @AashiqRamachandran, Metadata is not required for hunting queries within solution.